### ISSUE TITLE
Support addr:place addressing

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/id-tagging-schema#353]
```
addr:housenumber=10
addr:place=Village without street names
```

is a valid tagging, in widespread use, not supported by iD

![screen](https://user-images.githubusercontent.com/899988/149309663-dac63fa9-890a-4cb5-83f7-01114f8e8cb5.png)

When used iD encourages to add not actually missing data.

Ideally it would be shown as "Address without street name" or something, at least when tagged already

https://taginfo.openstreetmap.org/keys/addr%3Aplace
https://wiki.openstreetmap.org/wiki/Key%3Aaddr%3Aplace

See also https://github.com/openstreetmap/iD/issues/2898

**Comment by IgorEliezer** [comment]
There should be a drop-down menu where you can choose an address format valid for the region/BBOX. Something like:

_Select an address format:_
* Urban: House number, street, suburb, postcode, city
* Urban: No house number*, street, suburb, postcode, city _(POI needs a name)_
* Urban: Suburb/ward, cityblock, house number, postcode, city
* By milestone: Road, milestone, postcode, city
* Minor rural settlement: House name, road, postcode, city
* Hamlet: Place, house number, postcode
* Farms: Place, postcode

\* -- adds `nohousenumber=yes` tag

**Comment by tyrasd** [comment]
I would definitely like to improve this! It will however require a delicate approach since addresses are quite a complex matter 

Btw, I had proposed this one myself back in 2014 in https://github.com/openstreetmap/iD/pull/2246#issuecomment-45094187 :sweat_smile:

**Comment by 1ec5** [comment]
As seen in the screenshot in https://github.com/openstreetmap/id-tagging-schema/issues/353#issue-1101480704, the Address field is already capable of being laid out like a printed address on an envelope. It also varies by country. What isn’t currently supported is a mechanism for toggling between different formats relevant within a single country or conditionally showing one textbox based on another textbox being filled in. This capability would be useful regardless of `addr:place` in some countries like Vietnam that have distinct urban and rural address formats (and distinct mappings to OpenStreetMap tags).

**Comment by matkoniecz** [comment]
> JOSM's Address preset dialogue puts addr:place between addr: street and addr:city. There is no message in the dialogue to say that you should complete street or place but not both. This leads to people thinking addr:place is for suburbs.

Have you already reported this issue to JOSM developers? (or is there an open issue created by someone else?)

**Comment by RobJN** [comment]
>the Address field is already capable of being laid out like a printed address on an envelope. It also varies by country. What isn’t currently supported ...

I hadn't seen this as an envelope but I had realised the positioning was done to aid mappers. And I agree, it is all the bits that are missing that add complexity. For example in the UK one line of the address would be addr:housenumber + " " + addr:street. But if you add a addr:place tag then it becomes two lines showing addr:housenumber + " " + addr:place on one line, with addr:parentstreet on the second (yes, we have an addr:parentstreet tag in the UK because some addressees have a child+parent relationship type concept here).

> Have you already reported this issue to JOSM

Yes. https://josm.openstreetmap.de/ticket/21859#ticket

**Comment by bgo-eiu** [comment]
What would need to be implemented in order to make this happen? The two situations I can imagine are one where you select a format for a country, or one where you can just add more fields to it like [+] Block Number and so on. I'm curious because I've noticed recently how often addresses get mapped with things that are not street names in the street name field.

It seems like something flexible that doesn't assume too much about what sort of addresses a user is mapping would be the simplest. Not everywhere even has a ruleset about what what addresses should look like, so I think options like "urban address" and "village" address could potentially complicate things when the user likely knows which fields are relevant and which aren't as long as they're all options. Otherwise you could end up trying to account for things like "gas station in the Himalayas" address (usually has a milestone address) or "houseboat address" (might have a marina name in it as addr:place) and so on.

**Comment by Marc-marc-marc** [comment]
is it possible to at least show addr:place if set ? this would avoid [users filling addr:street because it is empty](https://www.openstreetmap.org/way/213743631/history)
is it possible to issue an validator error if addr:street + addr:place ?
is it possible to grey out/hide addr:postcode addr:city by country ? or retrieve the info from the encompassing relations ? this will avoid having users thinking the info is missing

**Comment by RobJN** [comment]
>Keep in mind that this is a valid combination despite the wiki.   

This is what I used to think too but not anymore. This year we had a very long discussion with the UK community and learnt about the history of addr:place (the tag didn't originate from the UK mapping community). It is for use when there is no addr:street. So certainly in many communities the validation rule proposed should be implemented.

As for the address example above, I'm not familiar with addresses in Bulgaria. We decided to respect the original intent of addr:place and not use it with addr:street. Through discussion we came up with other solutions. Should you wish to do the same then community discussion is the way forward. Of course if the local community is aware of this and has decided to use addr:place anyway, then fine but that shouldn't stop validation checks flagging it for most other countries.

For more details see the conclusions of the UK discussion here: https://wiki.openstreetmap.org/wiki/Addresses_in_the_United_Kingdom

**Comment by Dimitar5555** [comment]
> Of course if the local community is aware of this and has decided to use addr:place anyway, then fine but that shouldn't stop validation checks flagging it for most other countries.

I agree with you. Such cases are an exception to the rule. I think we already disscused it in the community and found some solution but I can't find the thread right now.

**Comment by lonvia** [comment]
This double addressing system is wide-spread in areas of the former Austria-Hungarian empire. (I presume that this is what @Dimitar5555 refers to although it is hard to say without knowing where the example comes from.) It is usually tagged in OSM using [`addr:streetnumber`](https://wiki.openstreetmap.org/wiki/Key:addr:streetnumber) + `addr:street` together with [`addr:conscriptionnumber`](https://wiki.openstreetmap.org/wiki/Key:addr:conscriptionnumber) + `addr:place`. It is indeed the exception where `addr:street` and `addr:place` appear together on the same address node. I'm not aware of any other exceptions.

**Comment by Dimitar5555** [comment]
@lonvia it's a Soviet era panel building. `addr:place` refers to the residential complex in which it is located. `addr:streetnumber` is used to represent the building number inside that complex (so you can have two `addr:streetnumber=бл. 1` as long as they have different values for `addr:place`). This particular building also has a street number which cause that collision. The cadastral map has it registered as `Residential Complex "Hipodruma", 1 Urvich St. 1, building №121`...

**Comment by Dimitar5555** [comment]
Is there any reason for keeping this issue open? It seems like it has been resolved.

**Comment by matkoniecz** [comment]
I will need to look into is it sign of bad data.

EDIT: added place= in https://www.openstreetmap.org/changeset/184640526

**Comment by matkoniecz** [comment]
fixed!
